controllers/check_client_compatibility.go

```diff
@@ -38,7 +38,7 @@ type checkClientCompatibility struct{}
 // reconcile runs the reconciler's work.
 func (c checkClientCompatibility) reconcile(_ context.Context, r *FoundationDBClusterReconciler, cluster *fdbv1beta2.FoundationDBCluster) *requeue {
 	logger := log.WithValues("namespace", cluster.Namespace, "cluster", cluster.Name, "reconciler", "checkClientCompatibility")
-	if !cluster.Status.Configured {
+	if !cluster.Status.Configured && !cluster.IsBeingUpgraded() {
```

Comment:
Does it mean we should still do the client compatibility check for version compatible upgrades?

Reply:
A few lines below we already have:

```go
	if version.IsProtocolCompatible(runningVersion) {
		return nil
	}
```

So we will skip the check for version compatible upgrades. The only modification here is to ignore this reconciler is we are not upgrading.

Reply:
Ok.